src/SIL.XForge.Scripture/Services/SFProjectService.cs

```diff
@@ -455,28 +454,27 @@ await RealtimeService
     }
 
     /// <summary> Get the link sharing key for a project if it exists, otherwise create a new one. </summary>
-    public async Task<string> GetLinkSharingKeyAsync(string curUserId, string projectId, string role)
+    public async Task<string> GetLinkSharingKeyAsync(
+        string curUserId,
+        string projectId,
+        string role,
+        string shareLinkType
+    )
     {
         SFProject project = await GetProjectAsync(projectId);
         if (!CanUserShareRole(curUserId, project, role))
             throw new ForbiddenException();
 
+        bool isProjectAdmin = IsProjectAdmin(project, curUserId);
+
         string[] availableRoles = new Dictionary<string, bool>
         {
             {
                 SFProjectRole.CommunityChecker,
-                project.CheckingConfig.CheckingEnabled
-                    && project.CheckingConfig.ShareEnabled
-                    && project.CheckingConfig.ShareLevel == CheckingShareLevel.Anyone
+                project.CheckingConfig.CheckingEnabled && (isProjectAdmin || project.CheckingConfig.ShareEnabled)
             },
-            {
-                SFProjectRole.SFObserver,
-                project.TranslateConfig.ShareEnabled && project.TranslateConfig.ShareLevel == TranslateShareLevel.Anyone
-            },
-            {
-                SFProjectRole.Reviewer,
-                project.TranslateConfig.ShareEnabled && project.TranslateConfig.ShareLevel == TranslateShareLevel.Anyone
-            }
+            { SFProjectRole.SFObserver, isProjectAdmin || (project.TranslateConfig.ShareEnabled) },
```

Comment:
Unnecessary brackets around `(project.TranslateConfig.ShareEnabled)` on this line and the one below it.